src/compiler/checker.ts

```diff
@@ -23258,6 +23263,7 @@ namespace ts {
 
         function checkExportSpecifier(node: ExportSpecifier) {
             checkAliasSymbol(node);
+            collectLinkedAliases(node.propertyName || node.name, /*setVisibility*/ true); // Collect linked aliases to set visibility links

```

Comment:
you only need to do this if `--declaration` is set..

Reply:
done